### PR TITLE
Add GCD algorithm to math library

### PR DESCRIPTION
**Pull request by alnoki**
@davidiw @movekevin @junkil-park 

Add to the `u64` and `u128` math libraries an implementation of Euclid's algorithm for computing the greatest common divisor, along with tests

### CODE REVIEW

File: aptos-move/framework/aptos-stdlib/sources/math64.move [hunk 1]
```diff
@@ -26,6 +26,17 @@ module aptos_std::math64 {
         }
     }
 
+    /// Return greatest common divisor of `a` & `b`, via the Euclidean algorithm.
+    public inline fun gcd(a: u64, b: u64): u64 {
```
Comment (junkil-park) [inline aptos-move/framework/aptos-stdlib/sources/math64.move:30]: Does this need to be a "inline" function?

**Comment by junkil-park** [comment]
Could you add `gcd` to `math128` as well for consistency?

**Comment by alnoki** [comment]
> Could you add `gcd` to `math128` as well for consistency?

@junkil-park 453cfae adds `gcd` function and test to `math128` library

**Review by movekevin — APPROVED** [review]

**Comment by alnoki** [comment]
@junkil-park anything else you'd like to see before an approving review and merge to `main`?

**Review by junkil-park — APPROVED** [review]

**Comment by junkil-park** [comment]
@alnoki, the lint test failed for this PR. Can you run `cargo build -p aptos-cached-packages` to regenerate the `.md` files?

**Comment by alnoki** [comment]
@junkil-park 9d340b7 rebuilds DocGen `.md` files

**Comment by alnoki** [comment]
@junkil-park while I'm not totally familiar with you CI stack, it appears that the TS SDK e2e test is failing due to a problem unrelated to this PR - is there something I can do to initiate a retry of that status check?

**Pull request merged**